packages/dds/tree/src/core/tree/pathTree.ts

```diff
@@ -59,6 +60,30 @@ export interface FieldUpPath<TUpPath extends UpPath = UpPath> {
 	readonly field: FieldKey; // TODO: Type information, including when in DetachedField.
 }
 
+/**
+ * An UpPath at the root of a detached tree.
+ */
+export interface DetachedUpPath extends UpPath {
+	/**
+	 * The DetachedNodeId of this path.
+	 */
+	readonly detachedNodeId: DetachedNodeId;
+}
+
+/**
+ * Given an {@link UpPath}, checks if it is detached.
+ */
+export function isDetachedUpPath(path: UpPath): path is DetachedUpPath {
+	return (path as DetachedUpPath).detachedNodeId !== undefined;
+}
+
+/**
+ * Given an {@link UpPath}, checks if it is detached and if so, returns it as a {@link DetachedUpPath}.
+ */
+export function getDetachedUpPath(path: UpPath): DetachedUpPath | undefined {
```

Comment:
`getDetachedUpPath` is only used by a single test file, and in that test file so I suggest moving it to that file for now.